### ISSUE TITLE
Multiple workflows fail

### ISSUE THREAD

**Issue by iparask** [radical-cybertools/radical.entk#364]
I'm running a use case where two workflows are submitted sequentially to EnTK. The code that does that looks like:
```python
        self._app_manager = AppManager(port=32773, hostname='localhost', name=args.name,
                            autoterminate=False, write_workflow=True)
        self._app_manager.workflow = set([discovery_pipeline])

        self._app_manager.run()
        self._app_manager.workflow = set([discovery_pipeline])

        self._app_manager.run()
```

The error I'm getting is:
```python
Traceback (most recent call last):
  File "/home/iparask/miniconda2/envs/iceberg/bin/iceberg", line 39, in <module>
    exec_obj.run()
  File "/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/iceberg/executor/seals.py", line 67, in run
    self._run_workflow()
  File "/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/iceberg/executor/seals.py", line 195, in _run_workflow
    self._app_manager.workflow = set([discovery_pipeline])
  File "/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/radical/entk/appman/appmanager.py", line 286, in workflow
    p._validate()
  File "/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/radical/entk/pipeline/pipeline.py", line 374, in _validate
    actual_value=self._state)
radical.entk.exceptions.ValueError: Value for attribute pipeline.0000 of object state incorrect. Expected value DESCRIBED, but got DONE.
```

The expected behavior is for the new pipelines to start from number `0001`.

**Comment by iparask** [comment]
@andre-merzky anything new on this? Thank you!

**Comment by iparask** [comment]
Even when the app manager closes and reopens, I'm still getting the same error. Is there something we can do? I can also work on it, but I would like some guidance.

**Comment by andre-merzky** [comment]
I am on it, finally...

**Comment by andre-merzky** [comment]
Can you please try with `autoterminate=False`, like this:

```
if __name__ == '__main__':

    appman = re.AppManager(hostname=hostname, port=port, autoterminate=False)
    appman.resource_desc = {'resource' : 'local.localhost',
                            'walltime' : 10,
                            'cpus'     : 2}

    appman.workflow = set([generate_pipeline()])
    appman.run()

    appman.workflow = set([generate_pipeline()])
    appman.run()

    appman.terminate()
```

**Comment by andre-merzky** [comment]
Apologies, Ioannis - I should have mentioned that you need `fix/issue_364` from RE, *and* `fix/issue_entk_364` from RP...

**Comment by iparask** [comment]
That is what I did in my last comment. Let me try the fix branches.

**Comment by iparask** [comment]
I'm getting the following error from setup.py:
```python
ERROR: Complete output from command /home/iparask/miniconda2/envs/iceberg/bin/python -u -c 'import setuptools, tokenize;__file__='"'"'/tmp/pip-req-build-0QCQZ7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-78RGWi --python-tag cp27:
  ERROR: /home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/setuptools/dist.py:481: UserWarning: The version specified ('0.70.0-v0.70.0-2-g4b12a43@fix-issue_364') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
    "details." % self.metadata.version
```

I will introduce the change in my local devel branch and I will let you know.

**Comment by iparask** [comment]
I'm getting the same error with devel as well.

**Comment by andre-merzky** [comment]
The `invalid version` error?  I have no idea what that means... :-(

**Comment by iparask** [comment]
I had some old example sessions, named `re....` and that was the cause. Let me try again.

**Comment by iparask** [comment]
Regardless the value of autoterminate, I'm still getting the same error.

**Comment by andre-merzky** [comment]
Did you also install the RP fix?  Can you send me the output of `radical-stack`, please?

**Comment by iparask** [comment]
There you go:
```
/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/radical/pilot
/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/radical/utils
/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/radical/entk
/home/iparask/miniconda2/envs/iceberg/lib/python2.7/site-packages/radical/saga

  python               : 2.7.16
  pythonpath           :
  virtualenv           : iceberg

  radical.entk         : 0.70.0-v0.70.0-2-g4b12a43@fix-issue_364
  radical.pilot        : 0.70.3-v0.70.3-114-gd5749c8@fix-issue_entk_364
  radical.saga         : 0.70.0
  radical.utils        : 0.70.0
```

**Comment by andre-merzky** [comment]
Thanks.  Well, that stack works for me with the example I posted.  Can you check if that exact example works for you?  I'll attach the complete code below.  If it does, can you provideinstructions on how to use your application?
[issue_364.py.gz](https://github.com/radical-cybertools/radical.entk/files/3503418/issue_364.py.gz)

**Comment by iparask** [comment]
The example runs locally, and I'll give a try on Bridges. You can find my application [here](https://github.com/iceberg-project/ICEBERG-middleware/blob/feature/seals_support/src/iceberg/executor/seals.py). I will setup the environment for you and provide you with instructions how to run it. We can also have a meeting tomorrow, if you want (I'm in Europe).

**Comment by iparask** [comment]
@andre-merzky, I just realized a bug from my side that eluded both of us. The initial code submits twice the same pipeline object, `discovery_pipeline`:
```
        self._app_manager.workflow = set([discovery_pipeline])

        self._app_manager.run()
        self._app_manager.workflow = set([discovery_pipeline])
```

Then what I saw was correct, but probably the error message did not help and made us start searching for something and fixed something else (?!).

How do you want to proceed?

**Comment by andre-merzky** [comment]
How to proceed somewhat depends on if there is still a bug :-)  Is there?

**Comment by iparask** [comment]
It does not.
